Upstream-Status: Pending [Not submitted to upstream yet]
Signed-off-by: Vishnu Banavath <vishnu.banavath@arm.com>

From b126e250c2c6f2388fe3f30ee270dc9a891ad3a4 Mon Sep 17 00:00:00 2001
From: Vishnu Banavath <vishnu.banavath@arm.com>
Date: Sat, 11 Dec 2021 13:23:55 +0000
Subject: [PATCH] efi_loader: corstone1000: remove guid check from corstone1000 config option

Use generic fmp guid and no separte check is required for
CORSTONE1000 target.

Signed-off-by: Vishnu Banavath <vishnu.banavath@arm.com>

diff --git a/lib/efi_loader/efi_capsule.c b/lib/efi_loader/efi_capsule.c
index f018c740ab..681470762a 100644
--- a/lib/efi_loader/efi_capsule.c
+++ b/lib/efi_loader/efi_capsule.c
@@ -554,18 +554,18 @@ efi_status_t __efi_runtime EFIAPI efi_update_capsule(
 			log_err("EFI: A capsule has not enough data\n");
 			continue;
 		}
+		if (guidcmp(&capsule->capsule_guid,
+			&efi_guid_firmware_management_capsule_id)) {
+			log_err("Unsupported capsule type: %pUl\n", &capsule->capsule_guid);
+			ret = EFI_UNSUPPORTED;
+			goto out;
+		}
 
 		log_debug("Capsule[%d] (guid:%pUl)\n",
 			  i, &capsule->capsule_guid);
 
 #if CONFIG_IS_ENABLED(TARGET_CORSTONE1000)
 
-		if (guidcmp(&corstone1000_capsule_guid, &capsule->capsule_guid)) {
-			ret = EFI_INVALID_PARAMETER;
-			log_err("Corstone1000: Invalid capsule GUID\n");
-			goto out;
-		}
-
 		if (efi_size_in_pages(capsule->capsule_image_size) >
 		    CORSTONE1000_CAPSULE_BUFFER_SIZE) {
 			log_err("Corstone1000: Capsule data size exceeds the shared buffer size\n");
@@ -591,15 +591,7 @@ efi_status_t __efi_runtime EFIAPI efi_update_capsule(
 		goto out;
 
 #endif
-
-		if (!guidcmp(&capsule->capsule_guid,
-			     &efi_guid_firmware_management_capsule_id)) {
-			ret  = efi_capsule_update_firmware(capsule);
-		} else {
-			log_err("EFI: not support capsule type: %pUl\n",
-				&capsule->capsule_guid);
-			ret = EFI_UNSUPPORTED;
-		}
+		ret  = efi_capsule_update_firmware(capsule);
 
 		if (ret != EFI_SUCCESS)
 			goto out;
-- 
2.17.1

